Assets/Scripts/Data Model/RasterLayer.cs

```diff
@@ -258,28 +260,68 @@ public float SampleIntensityBilinear(float u, float v)
 		return viewingRaster.GetPixelBilinear(u, v).r;
 	}
 
+	public float? GetRasterValueAt(Vector2 worldPosition)
+	{
+		float rasterValue = GetValueAt(worldPosition).r * rasterValueToEntityValueMultiplier;
+		// note MH: below cutoff value should not be mapped to a entity type
+		if (rasterValue < rasterObject.layer_raster_minimum_value_cutoff)
+		{
+			return null;
+		}
+		return rasterValue;
+	}
+
+	public float? GetRasterValueAt(int rasterSpaceX, int rasterSpaceY)
+	{
+		float rasterValue = viewingRaster.GetPixel(rasterSpaceX, rasterSpaceY).r * rasterValueToEntityValueMultiplier;
+		// note MH: below cutoff value should not be mapped to a entity type
+		if (rasterValue < rasterObject.layer_raster_minimum_value_cutoff)
+		{
+			return null;
+		}
+		return rasterValue;
+	}
+
+	[CanBeNull]
 	public EntityType GetEntityTypeForRasterAt(Vector2 worldPosition)
 	{
-		float rasterValue = GetValueAt(worldPosition).r * RASTER_VALUE_TO_ENTITY_VALUE_MULTIPLIER;
-		return GetEntityTypeForRasterValue(rasterValue);
+		float? rasterValue = GetRasterValueAt(worldPosition);
+		if (null == rasterValue)
+		{
+			return null;
+		}
+		return GetEntityTypeForRasterValue(rasterValue.Value);
 	}
 
+	[CanBeNull]
 	public EntityType GetEntityTypeForRasterAt(int rasterSpaceX, int rasterSpaceY)
 	{
-		float rasterValue = viewingRaster.GetPixel(rasterSpaceX, rasterSpaceY).r * RASTER_VALUE_TO_ENTITY_VALUE_MULTIPLIER;
-		return GetEntityTypeForRasterValue(rasterValue);
+		float? rasterValue = GetRasterValueAt(rasterSpaceX, rasterSpaceY);
+		if (null == rasterValue)
+		{
+			return null;
+		}
+		return GetEntityTypeForRasterValue(rasterValue.Value);
 	}
 
-	private EntityType GetEntityTypeForRasterValue(float rasterValue)
+	[CanBeNull]
+	public EntityType GetEntityTypeForRasterValue(float rasterValue)
 	{
-		EntityType result = entityTypesSortedByValue[0];
+		EntityType previousType = entityTypesSortedByValue[0];
+		EntityType result = previousType;
 		for (int i = 1; i < entityTypesSortedByValue.Count; ++i)
 		{
 			EntityType nextType = entityTypesSortedByValue[i];
 			if (nextType.value > rasterValue)
 			{
+				float lerp = Mathf.Clamp01((float)(rasterValue - previousType.value) / (float)(nextType.value - previousType.value));
```

Comment:
instead of just picking entity type lowBound. We now pick the one "closest" to the value. Otherwise their is a mismatch between the interpolated color and its entity type